### PR TITLE
images: cleanup after package installs in all images

### PR DESCRIPTION
**Pull request by idealhack**
- clean up the apt cache by removing `/var/lib/apt/lists`
- remove explicit `apt-get clean` which is not required

ref: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/

fixes https://github.com/kubernetes/test-infra/issues/7520

/assign @BenTheElder @dims 

(oh I should say `/cc` instead)

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Is it really suggested to `rm` instead of letting `apt` handle it?

**Comment by BenTheElder** [comment]
AFAIK apt will not remove the lists, outside of a docker container you normally wouldn't want to remove these. you can `apt clean` but that has different coverage.

Kubernetes's base image ships a small utility to use as an apt drop-in replacement for this purpose https://github.com/kubernetes/kubernetes/blob/master/build/debian-base/clean-install

File: planter/Dockerfile [hunk 1]
```diff
@@ -35,8 +35,8 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     unzip \
     wget \
     zip \
-    zlib1g-dev \
-    && apt-get clean && \
+    zlib1g-dev && \
+    rm -rf /var/lib/apt/lists/* \
```
Comment (ixdy) [inline planter/Dockerfile]: this is missing a `&&` at the end

Comment (idealhack) [reply planter/Dockerfile]: thanks for catching that! I'll open a follow up PR

**Pull request merged**